File: csharp/booking-up-for-beauty/BookingUpForBeauty.cs
using System;

static class Appointment
{
    public static DateTime Schedule(string appointmentDateDescription)
    {
        bool parsed = DateTime.TryParseExact(appointmentDateDescription, "M/d/yyyy HH:mm:ss",null, System.Globalization.DateTimeStyles.None, out DateTime dt)
        || DateTime.TryParseExact(appointmentDateDescription, "MMMM d, yyyy HH:mm:ss",null, System.Globalization.DateTimeStyles.None, out dt)
        || DateTime.TryParseExact(appointmentDateDescription, "dddd, MMMM d, yyyy HH:mm:ss",null, System.Globalization.DateTimeStyles.None, out dt);
    
        return dt;
    }

    public static bool HasPassed(DateTime appointmentDate) => appointmentDate < DateTime.Now;
    

    public static bool IsAfternoonAppointment(DateTime appointmentDate) => appointmentDate.Hour > 11 && appointmentDate.Hour < 18;
    

    public static string Description(DateTime appointmentDate) => $"You have an appointment on {appointmentDate:M/d/yyyy h:mm:ss tt}.";

    public static DateTime AnniversaryDate() => new DateTime(DateTime.Now.Year,9,15,0,0,0);
    
}

File: fsharp/booking-up-for-beauty/BookingUpForBeauty.fs
module BookingUpForBeauty

// The following line is needed to use the DateTime type
open System
open System.Globalization

let epoch = DateTime(2012, 9,15)

let schedule (appointmentDateDescription: string): DateTime = DateTime.ParseExact(appointmentDateDescription, [|"M/dd/yyyy HH:mm:ss";"MMMM d, yyyy HH:mm:ss"; "dddd, MMMM d, yyyy HH:mm:ss" |], CultureInfo.InvariantCulture, DateTimeStyles.None)

let hasPassed (appointmentDate: DateTime): bool = appointmentDate < DateTime.Now

let isAfternoonAppointment (appointmentDate: DateTime): bool =   appointmentDate.Hour > 11 && appointmentDate.Hour < 18

let description (apt: DateTime): string = $"You have an appointment on {apt:``M/d/yyyy h:mm:ss tt``}."

let anniversaryDate(): DateTime = DateTime(DateTime.Now.Year, epoch.Month, epoch.Day)
